test/Shared/UserManagerTestBase.cs

```diff
@@ -83,15 +83,10 @@ protected RoleManager<TRole> CreateRoleManager(object context = null, IServiceCo
         protected abstract void AddUserStore(IServiceCollection services, object context = null);
         protected abstract void AddRoleStore(IServiceCollection services, object context = null);
 
-        protected TUser CreateTestUser(string namePrefix = "")
-        {
-            return new TUser() { UserName = namePrefix + Guid.NewGuid().ToString() };
-        }
+        protected abstract TUser CreateTestUser(string username = "", string email = "", string phoneNumber = "",
+            bool lockoutEnabled = false, DateTimeOffset? lockoutEnd = null);
```

Comment:
There are some tests that need to remember the usernames for the next step, for example
```
var user = CreateUser(username)
result = await manager.FindByUserName(username)
```

For the above scenario I need to have the username at hand. Changing to namePrefix, it makes it tough to maintain the username. Earlier this was done using user.Username so it back to the same problem again. Can the overload take both username and flag that if true, uses what is passed ?